### DISCUSSION TITLE
ignore URL normalization for package name when adding dependencies.

### DISCUSSION THREAD

**Discussion by am3rb3ar** [pdm-project/pdm#1483] [Q&A] [answered]
Question (sorry if this is the wrong place to post this) - 

Is there a way to stop the normalization (ala [pep 503](https://peps.python.org/pep-0503/)] of package names when using `pdm update` or `pdm add`?

I.E. I'm trying to add `ns.utils` from a local artifactory repo.
PDM is sending a request to `artifactory/ns-utils` but the local server has this as `artifactory/ns.utils`.

Thanks in advanced.

**Comment by frostming** [comment, marked as answer]
It is a MUST:

> Repositories MAY redirect unnormalized URLs to the canonical normalized URL (e.g. /Foobar/ may redirect to /foobar/), however clients MUST NOT rely on this redirection and MUST request the normalized URL.

We just implement it as the standard and your local repo should be fixed to follow the simple index PEP.

**Reply by am3rb3ar** [reply]
That's what I thought, but just needed the confirmation - super thanks for the answer and the project!
